### ISSUE TITLE
External dependecies in 3.1.1.

### ISSUE THREAD

**Issue by Ayve1** [mozmorris/react-webcam#165]
Hello, 
I am using react-webcam since version 1.1.1 and it is great, but when i recently updated it to 3.1.1 I got problem with my Content-Security-Policy in Nginx.

react-webcam.js:271 Uncaught EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self' 'unsafe-inline'.

There should be external dependency in newer version that I have to add and I have no idea where to find it. Can you please help me with this problem?

**Comment by mozmorris** [comment]
Use the production version eg. `react-webcam.min.js`

https://unpkg.com/react-webcam/dist/react-webcam.min.js

**Comment by Ayve1** [comment]
I don't understand this answer, because I just use normal import in my project like: 
`import Webcam from "react-webcam";`
Please explain what do you mean by using production version or how to do it.

**Comment by mozmorris** [comment]
@Ayve1 okay let me try to help you.

Are you seeing the error in development? What about production?

**Comment by Ayve1** [comment]
Let me explain my situation as easy I can tell.
I am using Create react app for react project that is using your react-webcam.
To use development I am using: `react-scripts start`, which doesn't need Nginx. 
To use production I am using: `react-scripts build`. This command provide me build version of my project that I serve by Nginx to web.
So my project is on production.
Nginx configuration has optional function that provides more security which is called Content-Security-Policy and because I use this I have to write all external sites dependencies that I need in my project as exceptions. 
For example if I use google analitycs I have to add "https://www.google-analytics.com" in my nginx configuration.

In conclusion react-webcam version 3.1.1 needs something from other site that i have to include as exception in my Nginx configuration and because of this I can't use properly this version.

**Comment by mozmorris** [comment]
@Ayve1 thank you for the detail description.

It's not that react-webcam need something from another site, it's that your own domain needs adding as an exception as `react-webcam.js` contains `eval` in the code.

But I think you've found another problem tbh. When you create a production bundle, the eval shouldn't be generated.

I'm looking into this issue.

**Comment by Ayve1** [comment]
Okay, I understand everything now.
Thank you for your answer and I am looking forward for news about this issue.

**Comment by mozmorris** [comment]
@Ayve1 could you please test a new version for me?

`npm install react-webcam@next` or `yarn add react-webcam@next`

Does it still work correctly in development, and does it fix your issue in production?

**Comment by Ayve1** [comment]
@mozmorris  sorry for such delay in response.

Everything works fine now on production. I had no problem with Content-Security-Policy after serving site through Nginx.
On development I didnt have this problem but anyway I tested it and looks fine too.

Thanks for your hard work :)

**Comment by mozmorris** [comment]
Thanks for helping us test this @Ayve1 

The new version has been published as `4.0.0` so you may wish to update your `package.json`.

All the best 👍
